Controller/Search.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace Xfrocks\Api\Controller;
+
+use XF\Mvc\ParameterBag;
+use XF\Mvc\Entity\Entity;
+use Xfrocks\Api\Util\PageNav;
+
+class Search extends AbstractController
+{
+    public function actionGetIndex()
+    {
+        $data = [
+            'links' => [
+                'posts' => $this->buildApiLink('search/posts'),
+                'threads' => $this->buildApiLink('search/threads')
+            ]
+        ];
+
+        return $this->api($data);
+    }
+
+    public function actionGetResults(ParameterBag $params)
+    {
+        /** @var \XF\Entity\Search $search */
+        $search = $this->assertViewableEntity('XF:Search', $params->search_id);
+        if ($search->user_id !== \XF::visitor()->user_id) {
+            return $this->notFound();
+        }
+
+        $params = $this
+            ->params()
+            ->definePageNav();
+
+        list($perPage, $page) = $params->filterLimitAndPage();
+
+        $searcher = $this->app()->search();
+        $resultSet = $searcher->getResultSet($search->search_results);
+
+        $resultSet->sliceResultsToPage($page, $perPage);
+
+        if (!$resultSet->countResults()) {
+            return $this->error(\XF::phrase('no_results_found'), 400);
+        }
+
+        /** @var \Xfrocks\Api\ControllerPlugin\Search $searchPlugin */
+        $searchPlugin = $this->plugin('Xfrocks\Api:Search');
+
+        $data = [
+            'data_total' => $search->result_count,
+            'data' => $searchPlugin->prepareSearchResults($resultSet->getResults())
+        ];
+
+        PageNav::addLinksToData($data, $params, $data['data_total'], 'search/results', $search);
+
+        return $this->api($data);
+    }
+
+    public function actionPostThreads()
+    {
+        if (!\XF::visitor()->canSearch($error)) {
+            return $this->noPermission($error);
+        }
+
+        $params = $this
+            ->params()
+            ->define('q', 'str', 'query to search for')
+            ->define('forum_id', 'uint', 'id of the container forum to search for contents')
```

Comment:
Cái này là POST `/search/threads` rồi nên chắc chỉ cần ghỉ là `forum id to filter` là đủ.